sdk/keyvault/keyvault-certificates/src/lro/create/operation.ts

```diff
@@ -0,0 +1,161 @@
+   */
+  certificatePolicy: CertificatePolicy;
+  /**
+   * @member {CreateCertificatePollOperationProperties} [createCertificateOptions] The optional parameters that will be used to create the certificate
+   */
+  createCertificateOptions: CreateCertificateOptions;
+  /**
+   * @member {CreateCertificatePollOperationProperties} [client] An instance of the CertificatesClient class
+   */
+  client: CertificatesClientInterface;
+  /**
+   * @member {CreateCertificatePollOperationProperties} [initialResponse] The initial response received the first time the service was reached by the operation's update function
+   */
+  initialResponse?: CertificateOperation;
+  /**
+   * @member {CreateCertificatePollOperationProperties} [previousResponse] The previous response received the last time the service was reached by the operation's update function
+   */
+  previousResponse?: CertificateOperation;
+}
+
+/**
+ * @interface
+ * An interface representing a create certificate's poll operation
+ */
+export interface CreateCertificatePollOperation
+  extends PollOperation<CreateCertificatePollOperationProperties, CertificateOperation> {}
+
+/**
+ * @summary Reaches to the service and updates the create certificate's poll operation.
+ * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+ */
+async function update(
+  this: CreateCertificatePollOperation,
+  options: {
+    abortSignal?: AbortSignalLike;
+    fireProgress?: (properties: CreateCertificatePollOperationProperties) => void;
+  } = {}
+): Promise<CreateCertificatePollOperation> {
+  const {
+    name,
+    certificatePolicy,
+    createCertificateOptions,
+    client,
+    initialResponse,
+    previousResponse
+  } = this.properties;
+  const requestOptions = this.properties.createCertificateOptions.requestOptions || {};
+  if (options.abortSignal) {
+    requestOptions.abortSignal = options.abortSignal;
+  }
+
+  let response: CertificateOperation;
+  const doFinalResponse = previousResponse && previousResponse.status !== "inProgress";
+
+  if (!initialResponse) {
+    await client.createCertificate(name, certificatePolicy, createCertificateOptions);
+    response = await client.getCertificateOperation(name, requestOptions);
+    this.properties.initialResponse = response;
+  } else if (doFinalResponse) {
+    response = await client.getCertificateOperation(name, requestOptions);
+    this.state.completed = true;
+    this.state.result = response;
+  } else {
+    response = await client.getCertificateOperation(name, requestOptions);
+  }
+
+  const properties: CreateCertificatePollOperationProperties = {
+    ...this.properties,
+    previousResponse: response
+  };
+
+  // Progress only after the poller has started and before the poller is done
+  if (!(!initialResponse || doFinalResponse) && options.fireProgress) {
```

Comment:
nit: the double negative harms readability for me

```suggestion
  if (initialResponse && !doFinalResponse && options.fireProgress) {
```